docs/low-code-spark/configuration.md

```diff
@@ -9,76 +9,103 @@ tags:
-   :::note
-   Each name-value pair has to be first defined in the `Common` tab. This can then be overridden in the individual Fabric tabs.
-   ![Configurations - Common](./img/configs_common.png)
-   :::
-2. **Spark Configuration**: Runtime Spark configurations as name-value pairs.
-   :::note
-   The name-value pairs will be set inside the Spark runtime configurations as `spark.conf.set(name, value)`
-   ![Configurations - Spark](./img/configs_spark.png)
-   This will be compiled as:
+Prophecy IDE allows you to define three kinds of configurations:
 
-   ```python
-   spark.conf.set("spark.sql.autoBroadcastJoinThreshold", "10485760")
-   ```
+## Spark Configuration
 
-   :::<br/>
+Runtime Spark configurations as name-value pairs. The name-value pairs will be set inside the Spark runtime configurations as `spark.conf.set(name, value)`
 
-3. **Hadoop Configuration**: Hadoop configurations as name-value paris.
-   :::note
-   The name-value pairs will be set inside the Hadoop configuration as `spark.sparkContext.hadoopConfiguration.set(name, value)`
-   ![Configurations - Spark](./img/configs_hadoop.png)
-   This will be compiled as:
+![Configurations - Spark](./img/configs_spark.png)
 
-   ```python
-   spark.sparkContext.hadoopConfiguration.set("fs.s3a.access.key", "my_access_key")
-   ```
+## Hadoop Configuration
 
-   :::<br/>
+Hadoop configurations as name-value pairs. The name-value pairs will be set inside the Hadoop configuration as `spark.sparkContext.hadoopConfiguration.set(name, value)`
 
----
+![Configurations - Spark](./img/configs_hadoop.png)
+
+## Pipeline Configuration
+
+Config values which can be set at Pipeline level and then be accessed inside any component in the Pipeline. [Multiple instances](#pipeline-configuration-instances)
+of configuration can be created per Pipeline.
+![Configurations - Common](img/config-pipeline-eg1.png)
+
+Syntax for using configuration inside Gems:
+
+For visual language SQL : `'$config_name'`
+For visual language Scala/Python : `Config.config_name`
+
+For using Spark expression with visual language SQL : `expr('$config_name')`
+For using Spark expression with visual language Scala/Python : `expr(Config.config_name)`
+
+## Examples for Pipeline level configurations
+
+Now let's use the [above defined configurations](#pipeline-configuration) in the below Pipeline.
+![Pipeline view](img/config-pipeline-view-eg.png)
+
+### Using Config in limit Gem
+
+In the below image `'$num_top_customers'` is fetching the integer value defined in configurations.
+
+![Config Limit Example](img/config-pipeline-limit-eg.png)
+
+### Using Spark-expression Config type in Gem
+
+Here we have used Spark expression directly from our config value to populate a column.
+
+In the below image `amounts` -> `expr('$test_expression')` is coming from configuration type defined as `Spark-expression`
```

Comment:
Add numbers/letters to the image to draw the reader's attention to them, then put a table under the image with the explanation. See the Concepts section of documents for some examples.